framework/tests/src/tests/signatureinference/SignatureInferenceTestAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,162 @@
+import tests.signatureinference.qual.SiblingWithFields;
+import tests.signatureinference.qual.SignatureInferenceBottom;
+import tests.signatureinference.qual.Top;
+/**
+ * AnnotatedTypeFactory to test signaature inference using .jaif
+ * files.
+ * <p>
+ * The used qualifier hierarchy is straightforward and only intended for test
+ * purposes.
+ * 
+ * @author pbsf
+ */
+public class SignatureInferenceTestAnnotatedTypeFactory
+        extends
+            BaseAnnotatedTypeFactory {
+
+    private final AnnotationMirror PARENT = new AnnotationBuilder(
+            processingEnv, Parent.class).build();
+
+    public SignatureInferenceTestAnnotatedTypeFactory(BaseTypeChecker checker) {
+        super(checker);
+        postInit();
+        AnnotationMirror bottom = AnnotationUtils.fromClass(elements,
+                SignatureInferenceBottom.class);
+        addTypeNameImplicit(java.lang.Void.class, bottom);
+    }
+
+    @Override
+    protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+        return Collections.unmodifiableSet(new HashSet<Class<? extends Annotation>>(Arrays.asList(
+                    Parent.class, DefaultType.class, Top.class, Sibling1.class,
+                        Sibling2.class, SignatureInferenceBottom.class, SiblingWithFields.class)));
+    }
+    @Override
+    public TreeAnnotator createTreeAnnotator() {
+        ImplicitsTreeAnnotator implicitsTreeAnnotator = new ImplicitsTreeAnnotator(
+                this);
+        AnnotationMirror bottom = AnnotationUtils.fromClass(elements,
+                SignatureInferenceBottom.class);
+        implicitsTreeAnnotator.addTreeKind(
+                com.sun.source.tree.Tree.Kind.NULL_LITERAL, bottom);
+        implicitsTreeAnnotator.addTreeKind(
+                com.sun.source.tree.Tree.Kind.INT_LITERAL, bottom);
+
+        return new ListTreeAnnotator(new PropagationTreeAnnotator(this),
+                implicitsTreeAnnotator);
+    }
+
+    @Override
+    public QualifierHierarchy createQualifierHierarchy(MultiGraphFactory factory) {
+        return new SignatureTestQualifierHierarchy(factory);
+    }
+
+    /**
+     * Using a MultiGraphQualifierHierarchy to enable tests with Annotations
+     * that contain fields. @see SiblingWithFields.
+     * @author pbsf
+     *
+     */
+    protected class SignatureTestQualifierHierarchy extends
+                MultiGraphQualifierHierarchy {
+
+        public SignatureTestQualifierHierarchy(MultiGraphFactory f) {
+            super(f);
+        }
+
+        @Override
+        public AnnotationMirror leastUpperBound(AnnotationMirror a1,
+                AnnotationMirror a2) {
+            if ((AnnotationUtils.areSameByClass(a1, Sibling1.class)
+                    && AnnotationUtils.areSameByClass(a2, Sibling2.class)) 
+                    || (AnnotationUtils.areSameByClass(a1, Sibling2.class)
+                    && AnnotationUtils.areSameByClass(a2, Sibling1.class))
```

Comment:
This indentation is incorrect.  The entire expression should be to the right and below the open parenthesis.  You can fix that by moving "&&" under the "A" in "(AnnotationUtils.areSameByClass(a1, Sibling2.class)".  Same comment for other following lines.